### PR TITLE
[TRTLLM-11058][feat] Support Helix CP with GQA

### PR DESCRIPTION
**Pull request by brb-nv**
## Description

This MR generalizes Helix CP from MLA-only to standard GQA/MHA.

1. **Attention module & kernels:**
- Added helix functionality to base `Attention` class.
- Plumbed helix position offsets, inactive-rank flags, and softmax stats through the C++ kernel stack for GQA. Please note that even context kernels have been updated for testing though Helix is decode-only.
- Cleaned up `enable_helix_test` from MLA module as it's only for testing.

2. **Cache transmission:**
- Updated `splitKVCacheKernel` and `cacheFormatter` for CP-aware block distribution.

3. **Qwen3 as example:**
- Made changes to Qwen3 to serve as an example.
- We can probably avoid maintaining two mappings if we make MLP & MoE layers CP-aware. Deferred to a follow-up MR: https://jirasw.nvidia.com/browse/TRTLLM-11067.

## Test Coverage

```
$ pytest tests/unittest/_torch/modules/test_mla_helix.py -s -v
$ pytest tests/unittest/_torch/modules/test_mha_helix.py -s -v
$ TRTLLM_USE_UCX_KVCACHE=1 TLLM_LOG_LEVEL=INFO mpirun -n 8 ./tests/unit_tests/multi_gpu/cacheTransceiverTest --gtest_filter="AsymmetricCaseTestWithCPAndDPForGQA*/AsymmetricalCacheTestWithDP.TestCase/*"
$ TRTLLM_USE_UCX_KVCACHE=1 TLLM_LOG_LEVEL=INFO mpirun -n 8 ./tests/unit_tests/multi_gpu/cacheTransceiverTest --gtest_filter="AsymmetricCaseTest*WithCPForGQA/AsymmetricalCacheTest.TestCase/*"
$ ./tests/unit_tests/multi_gpu/cacheTransceiverTest --gtest_filter="targetTest.CacheStateNODPForGQAWithCP"
$ pytest tests/integration/defs/accuracy/test_disaggregated_serving.py::TestQwen3_8B::test_auto_dtype_with_helix -s -v
```

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

### CODE REVIEW

**Review by yechank-nvidia — APPROVED** [review]
LGTM on Qwen3's change. Also, this PR has CP tests on Qwen3. Thanks!

**Review by yizhang-nv — APPROVED** [review]
LGTM for torch.compile part

**Review by byshiue — COMMENTED** [review]
Approve for changes about Qwen.

**Review by byshiue — APPROVED** [review]

**Review by StanleySun639 — APPROVED** [review]

**Review by chuangz0 — APPROVED** [review]
approve for cache transceiver

File: cpp/tensorrt_llm/kernels/xqaDispatcher.cpp [hunk 1]
```diff
@@ -505,6 +507,7 @@ void XqaDispatcher::runImpl(
         tllmRunnerParams.customMaskOffsetsPtr = params.spec_decoding_bl_tree_mask_offset;
         tllmRunnerParams.firstSparseMaskOffsetsKvPtr = params.spec_bl_tree_first_sparse_mask_offset_kv;
         tllmRunnerParams.mSkipSoftmaxThresholdScaleFactor = params.skip_softmax_threshold_scale_factor;
+        tllmRunnerParams.softmaxStatsPtr = params.softmax_stats;
```
Comment (pengbowang-nv) [inline cpp/tensorrt_llm/kernels/xqaDispatcher.cpp]: I'd like to know the support status for Helix CP. I think we only want to support SM100 through trtllm-gen in TRTLLM attention backend, as we only have softmax_stats param passed here?

If so, I wonder what's the purpose of the modifications in `cpp/tensorrt_llm/kernels/decoderMaskedMultiheadAttention/decoderMaskedMultiheadAttentionTemplate.h` (MMHA kernels)? Thanks!

File: tensorrt_llm/_torch/attention_backend/trtllm.py [hunk 2]
```diff
@@ -593,7 +593,6 @@ def run(
                 self.v_head_dim,
                 self.mrope_rotary_cos_sin,
                 self.mrope_position_deltas,
-                mla_tensor_params,
```
Comment (pengbowang-nv) [inline tensorrt_llm/_torch/attention_backend/trtllm.py]: Just want to confirm. So we don't want to use flash-infer trtllm-gen attention kernel for Helix CP, and we only use the attention kernel by trtllm C++ binding? 

Also, as we're removing this support, do we have checks for situations where users set both _TRTLLM_ENABLE_TRTLLM_GEN_ATTENTION and helix parallel?

**Review by pengbowang-nv — APPROVED** [review]

**Review by mikeiovine — APPROVED** [review]
Signing off on torch module changes
